bundles/org.openhab.core.io.rest.auth/src/main/java/org/openhab/core/io/rest/auth/internal/AuthFilter.java

```diff
@@ -71,19 +78,49 @@
     protected static final String CONFIG_URI = "system:restauth";
     private static final String CONFIG_ALLOW_BASIC_AUTH = "allowBasicAuth";
     private static final String CONFIG_IMPLICIT_USER_ROLE = "implicitUserRole";
+    private static final String CONFIG_CACHE_EXPIRATION = "cacheExpiration";
 
     private boolean allowBasicAuth = false;
     private boolean implicitUserRole = true;
+    private Long cacheExpiration = 6L;
+
+    private ExpiringUserSecurityContextCache authCache = new ExpiringUserSecurityContextCache(
+            Duration.ofHours(cacheExpiration).toMillis());
+
+    static private byte[] RANDOM_BYTES = new byte[32];
+    static {
+        new Random().nextBytes(RANDOM_BYTES);
```

Comment:
If you change `RANDOM_BYTES` in to a non-static variable it can be initialized in the constructor.

Reply:
Right, by I preferred to initialize it statically. Is that an issue?

Reply:
It will probably work, but I ran into a nasty classloader issue due to static variables yesterday, so I'm a currently a bit less enthusiastic about seeing them being used. :wink:

Reply:
Ok, changed it!